integrations/telegram/src/index.ts

```diff
@@ -146,12 +147,33 @@ const integration = new bp.Integration({
       throw new Error('Handler received an empty from id')
     }
 
+    const userName = getUserNameFromTelegramUser(data.message.from as User)
+
     const { user } = await client.getOrCreateUser({
       tags: {
         [prefixedId]: `${userId}`,
       },
+      name: userName || userId,
     })
 
+    const userFieldsToUpdate = {
+      pictureUrl:
+        user.pictureUrl ||
+        (await getUserPictureDataUri({
```

Comment:
issue: We should do this call in a try/catch block and store it in a variable separately to log the error as a warning and return `null` if the image retrieval and download fails, to avoid blocking the processing of the message. Could be something simple like:

```ts
const pictureUrl = await getUserPictureDataUri(props).catch((err) => {
  logger.forBot().error(`Failed to retrieve image for Telegram user ${userId}: ${err.stack ?? err.message}`);
  return null;
})
```

Reply:
this is done already inside `getUserPictureDataUri`, why should I do it here as well ?

Reply:
My bad, you're right. Didn't see the try above lol.